esx_service/vmdk_ops.py

```diff
@@ -490,24 +490,27 @@ def get_vol_path(datastore, tenant_name=None):
     if os.path.isdir(path):
         # If the path exists then return it as is.
         logging.debug("Found %s, returning", path)
-        return path
+        return path, None
 
     if not os.path.isdir(dock_vol_path):
         # The osfs tools are usable for DOCK_VOLS_DIR on all datastores
         cmd = "{0} {1}".format(OSFS_MKDIR_CMD, dock_vol_path)
         rc, out = RunCommand(cmd)
         if rc != 0:
-            logging.warning("Failed to create %s", dock_vol_path)
-            return None
+            errMsg = "{0} creation failed - {1} on {2}".format(DOCK_VOLS_DIR, os.strerror(rc), datastore)
+            logging.warning(errMsg)
+            return None, err(errMsg)
     if tenant_name and not os.path.isdir(path):
+        # The mkdir command is used to create "tenant_name" folder inside DOCK_VOLS_DIR on "datastore"
         cmd = "{0} {1}".format(MKDIR_CMD, path)
         rc, out = RunCommand(cmd)
         if rc != 0:
-           logging.warning("Failed to create %s", path)
-           return None
+            errMsg = "Failed to initialize volume path {0}".format(path)
```

Comment:
Why not include error string here?
errMsg = "Failed to initialize volume path {0} :  {1}".format(path, os.strerror(rc))